salesforce/events.go

```diff
@@ -0,0 +1,277 @@
+	JwtFormulaSubject        string `json:"jwtFormulaSubject,omitempty"`
+	JwtIssuer                string `json:"jwtIssuer,omitempty"`
+	JwtSigningCertificate    string `json:"jwtSigningCertificate,omitempty"`
+	JwtTextSubject           string `json:"jwtTextSubject,omitempty"`
+	JwtValidityPeriodSeconds int    `json:"jwtValidityPeriodSeconds,omitempty"`
+	OauthRefreshToken        string `json:"oauthRefreshToken,omitempty"`
+	OauthScope               string `json:"oauthScope,omitempty"`
+	OauthToken               string `json:"oauthToken,omitempty"`
+	Password                 string `json:"password,omitempty"`
+	PrincipalType            string `json:"principalType,omitempty"`
+	Protocol                 string `json:"protocol,omitempty"`
+	Username                 string `json:"username,omitempty"`
+}
+
+// nolint:tagliatelle
+type NamedCredential struct {
+	FullName string                   `json:"FullName"`
+	Metadata *NamedCredentialMetadata `json:"Metadata"`
+
+	// below exist in response, but not in request
+	Id string `json:"Id,omitempty"`
+}
+
+func (n *NamedCredential) DestinationResourceName() string {
+	return fmt.Sprint("callout:", n.FullName)
+}
+
+// nolint: lll
+// https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_platformeventchannel.htm
+func (c *Connector) CreateEventChannel(ctx context.Context, channel *EventChannel) (*EventChannel, error) {
+	res, err := c.postToSFAPI(ctx, channel, "tooling/sobjects/PlatformEventChannel", "PlatformEventChannel")
+	if err != nil {
+		return nil, err
+	}
+
+	channel.Id = res.Id
+
+	return channel, nil
+}
+
+// nolint: lll
+// https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_platformeventchannelmember.htm
+func (c *Connector) CreateEventChannelMember(
+	ctx context.Context,
+	member *EventChannelMember,
+) (*EventChannelMember, error) {
+	res, err := c.postToSFAPI(ctx, member, "tooling/sobjects/PlatformEventChannelMember", "EventChannelMember")
+	if err != nil {
+		return nil, err
+	}
+
+	member.Id = res.Id
+
+	return member, nil
+}
+
+// nolint: lll
+// https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_eventrelayconfig.htm?q=EventRelayConfig
+func (c *Connector) CreateEventRelayConfig(
+	ctx context.Context,
+	cfg *EventRelayConfig,
+) (*EventRelayConfig, error) {
+	res, err := c.postToSFAPI(ctx, cfg, "/tooling/sobjects/EventRelayConfig", "EventRelayConfig")
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.Id = res.Id
+
+	return cfg, nil
+}
+
+// nolint: lll
+// https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_eventrelayconfig.htm?q=EventRelayConfig
+func (c *Connector) RunEventRelay(ctx context.Context, cfg *EventRelayConfig) error {
+	location, err := joinURLPath(c.BaseURL, "tooling/sobjects/EventRelayConfig", cfg.Id)
+	if err != nil {
+		return err
+	}
```

Comment:
You shouldn't need to manually join paths like that, the client (assuming it's set up with knowledge of `c.BaseURL`) will do that automatically.